include/boost/math/distributions/dirichlet.hpp

```diff
@@ -100,45 +92,33 @@ inline bool check_x(const char *function,
       return false;
     }
   }
-  return std::accumulate(x.begin(), x.end(), 0.0) <= 1.0;
+  return accumulate(x.begin(), x.end(), 0.0) <= 1.0;
```

Comment:
`0.0` parses as a double precision value, no matter what your `RealType` is. Use `RealType(0)`. Also, use `std::accumulate`, rather than `accumulate`, since you don't need argument-dependent lookup here.

Reply:
hmmm...